components/eamxx/src/physics/nudging/eamxx_nudging_process_interface.cpp

```diff
@@ -461,43 +505,45 @@ void Nudging::run_impl (const double dt)
     //       surface.
     // Here we change the int_state_view which represents the vertically interpolated fields onto
     // the simulation grid.
-    const int num_levs = m_num_levs;
-    Kokkos::parallel_for("correct_for_masked_values", policy,
-       	       KOKKOS_LAMBDA(MemberType const& team) {
-      const int icol = team.league_rank();
-      auto int_mask_view_1d  = ekat::subview(int_mask_view,icol);
-      auto tmp_state_view_1d = ekat::subview(tmp_state_view,icol);
-      Real fill_value;
-      int  fill_idx = -1;
-      // Scan top to surf and backfill all values near TOM that are masked.
-      for (int kk=0; kk<num_levs; ++kk) {
-        const auto ipack = kk / mPack::n;
-	const auto iidx  = kk % mPack::n;
-        // Check if this index is masked
-	if (!int_mask_view_1d(ipack)[iidx]) {
-	  fill_value = tmp_state_view_1d(ipack)[iidx];
-	  fill_idx = kk;
-	  for (int jj=0; jj<fill_idx; ++jj) {
-            const auto jpack = jj / mPack::n;
-	    const auto jidx  = jj % mPack::n;
-	    tmp_state_view_1d(jpack)[jidx] = fill_value;
-	  }
-	  break;
-	}
-      }
-      // Now fill the rest, the fill_idx should be non-negative.  If it isn't that means
-      // we have a column that is fully masked
-      for (int kk=fill_idx+1; kk<num_levs; ++kk) {
-        const auto ipack = kk / mPack::n;
-	const auto iidx  = kk % mPack::n;
-        // Check if this index is masked
-	if (!int_mask_view_1d(ipack)[iidx]) {
-	  fill_value = tmp_state_view_1d(ipack)[iidx];
-	} else {
-	  tmp_state_view_1d(ipack)[iidx] = fill_value;
-	}
-      }
-    });
+    if (m_src_pres_type != TIME_DEPENDENT_3D_HYBRID) {
```

Comment:
This is fine, but just to share a different perspective (for future reference): you could consider early return instead of a long if statement. It gets a bit hard to track nesting when there are several layers. Imho, something like this is clearer (in addition of only causing a 4 lines diff instead of ~40).
```c++
  if (m_src_pres_type == TIME_DEPENDENT_3D_HYBRID) {
    // Nothing left to do
    return;
  }
  
  const int num_levs = m_num_levs;
  ...
```
Again, what you did is fine. Just sharing this, in case it looks better to you.

Reply:
Thank you for telling me this! I don't want unnecessary diff either